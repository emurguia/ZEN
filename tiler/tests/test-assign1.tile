// This tests that assignment is generally working for int, float, string, bool

init {
	// Type declarations
	int a;
	int b;
	int c;

	string i;
	string j;

	float m;
	float n;

	bool x;
	bool y;

	tile(3,3);

	// Assignment of variables
	a = 42 - 17;
	b = 6 * 3;
	c = b + 64;
	m = 1.12341;
	n = 3.180; 
	i = "It's a me Mario";
	j = "Oh hello, Luigi";
	x = true;
	y = !(!x);

	iprint(a);
	iprint(b);
	iprint(c);

	sprint(i);
	sprint(j);

	fprint(m);
	fprint(n);

	if (x) {sprint("true");} else {sprint("false");}
	if (y) {sprint("true");} else {sprint("false");}

	close();

}