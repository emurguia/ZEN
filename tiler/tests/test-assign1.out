25
18
82
It's a me Mario
Oh hello, Luigi
1.123410
3.180000
true
true
The game window was successfully launched ...
The game window has been exited. Program exited.
Game over.